fix(integrations): Add `fixes <SHORT-ID>` pitfalls with squashed commits (#5684)
getsentry/sentry-docs: src/docs/product/integrations/source-code-mgmt/azure-devops/index.mdx

Fixes MYAPP-317
```

> When Sentry sees this, we’ll automatically annotate the matching issue with reference to the commit, and later, when that commit is part of a release, we’ll mark the issue as resolved. The commit must be associated with a release, if the commit is squashed Sentry will not know when the commit has been released and the issue may never be marked as a regression.

Reviewer: Does this rearranging of the sentence work? I want to make sure we're building the consequences of not associating the commit with a release. So first Sentry doesn't know about the commit/release and then there's a further consequence if it gets squashed. The way it's written currently, the cause and effect aren't super clear. If this works, then go ahead and do it for the other instances of this phrasing, but if not, ping me so we can figure something out.

```suggestion
When Sentry sees this, we’ll automatically annotate the matching issue with reference to the commit, and later, when that commit is part of a release, we’ll mark the issue as resolved. The commit must be associated with a release or Sentry won't know when the commit has been released and if the commit is squashed, the issue may never be marked as a regression.
```

> When Sentry sees this, we’ll automatically annotate the matching issue with reference to the commit, and later, when that commit is part of a release, we’ll mark the issue as resolved. The commit must be associated with a release or Sentry won't know when the commit has been released and if the commit is squashed, the issue may never be marked as a regression.

Reviewer: ```suggestion
When Sentry sees this, we’ll automatically annotate the matching issue with reference to the commit, and later, when that commit is part of a release, we’ll mark the issue as resolved. The commit must be associated with a release. Otherwise, if the commit is squashed, Sentry won’t know when the commit has been released, and the issue may never be marked as a regression.
```
